streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -165,14 +173,26 @@ public void closeSuspended(final boolean clean,
      */
     public List<ConsumerRecord<byte[], byte[]>> update(final TopicPartition partition,
                                                        final List<ConsumerRecord<byte[], byte[]>> records) {
+        if (records.isEmpty()) {
+            return Collections.emptyList();
+        }
+
         log.trace("Updating standby replicas of its state store for partition [{}]", partition);
-        final long limit = stateMgr.offsetLimit(partition);
+        long limit = offsetLimits.getOrDefault(partition, Long.MAX_VALUE);
```

Comment:
Since we do not initialize the offsetLimits, we would always get Long.MAX_VALUE for the first time right?

Reply:
We initialize the offset to 0 for changelog / source partitions in the constructor, here:

https://github.com/apache/kafka/pull/7238/files#diff-7012e86d489a291404d83f961ad40217R73